Ryujinx.HLE/HOS/Applets/SoftwareKeyboard/SoftwareKeyboardApplet.cs

```
@@ -76,6 +97,22 @@ private void Execute()
                 _keyboardConfig.StringLengthMax = 100;
             }
 
+            var args = new SoftwareKeyboardUiArgs
+            {
+                HeaderText = _keyboardConfig.HeaderText,
+                SubtitleText = _keyboardConfig.SubtitleText,
+                GuideText = _keyboardConfig.GuideText,
+                SubmitText = (!string.IsNullOrWhiteSpace(_keyboardConfig.SubmitText) ? _keyboardConfig.SubmitText : "OK"),
+                StringLengthMin = _keyboardConfig.StringLengthMin, 
+                StringLengthMax = _keyboardConfig.StringLengthMax,
+                InitialText = initialText
+            };
+
+            // Call the configured GUI handler to get user's input
+            _okPressed = _device.UiHandler.DisplayInputDialog(args, out _textValue);
```

> jduncanator: This should check UiHandler is not null, and leave the default _textValue if it is.

This comment is relevant again since I moved error handling out of applet code.
Not doing a null-check was intentional, because I feel adding a DummyHandler in the future is better than silently ignoring it here (at least until the code-base is stable)

---

Given that `UiHandler` _can_ be null, for whatever reason, it's appropriate to do an null check. You can just simply wrap it in an `if` statement and leave the default `_textValue` if it's null.

```
if(_device.UiHandler != null)
{
    // ...
}
```